DNN Platform/Library/Entities/Portals/PortalSettings.cs

```diff
@@ -699,15 +699,19 @@ public string GetProperty(string propertyName, string format, CultureInfo format
             var isPublic = true;
             switch (lowerPropertyName)
             {
+                case "scheme":
+                    propertyNotFound = false;
+                    result = SSLEnabled ? "https" : "http";

```

Comment:
The check in the code we're replacing is `portalSettings.SSLEnabled || portalSettings.SSLEnforced`.  I don't know that it hurts anything for it to remain also checking `SSLEnforced` (probably better to err on the side of more HTTPS rather than less).

Reply:
it should check for portalSettings.SSLEnabled && portalSettings.SSLEnforced - just SSLEnabled will deliver secured pages using http, if not called using https (e.g. if moved to a different location without https being bound to the IIS website.

Reply:
My thought was indeed to err on the side of https. Rapidly more and more technologies are refusing to work with plain http (e.g. mobile apps). So if a site has so much as allowed SSL, I suggest the default return of "scheme" should be https. Rather than waiting to see if it is enforced.

Reply:
Right, so `SSLEnabled || SSLEnforced` could be `true` more often than just `SSLEnabled` (though I'm not sure if there's any valid scenario where `SSLEnforced` is `true` while `SSLEnabled` is `false`)

Reply:
Looking at this, and the option values, it seems that you could have SSL Enforced = true, but SSL Enabled = false.  For this, I think we have to rely only on the `SSLEnabled` flag though, it may be a bit aggressive, but for current standards should be acceptable